packages/compiler/src/compiler/readMetadata.test.ts

```diff
@@ -253,22 +254,94 @@ describe('config', async () => {
     })
   })
 
-  it('does not compile the prompt as YAML when it is not the first element in the prompt', async () => {
-    const prompt = `
+  it('fails when the config is not at the top of the prompt', async () => {
```

Comment:
I thought we wanted to allow comments for the sample project

Reply:
product won't like it but i actually agree that frontmatter should be in... the front

Reply:
Oh nono, we allow comments. We just don't allow actual content like messages or text.